streams/src/test/java/org/apache/kafka/streams/integration/PauseResumeIntegrationTest.java

```diff
@@ -0,0 +1,380 @@
+        produceToInputTopics(INPUT_STREAM_2, STANDARD_INPUT_DATA);
+
+        assertNoLag(streamsNamedTopologyWrapper);
+
+        waitUntilStreamsHasPolled(streamsNamedTopologyWrapper, 2);
+        assertTopicSize(OUTPUT_STREAM_1, 3);
+        assertTopicSize(OUTPUT_STREAM_2, 3);
+
+        streamsNamedTopologyWrapper.resumeNamedTopology(TOPOLOGY1);
+        assertFalse(streamsNamedTopologyWrapper.isPaused());
+        assertFalse(streamsNamedTopologyWrapper.isNamedTopologyPaused(TOPOLOGY1));
+        assertTrue(streamsNamedTopologyWrapper.isNamedTopologyPaused(TOPOLOGY2));
+        awaitOutput(OUTPUT_STREAM_1, 3, COUNT_OUTPUT_DATA2);
+        assertTopicSize(OUTPUT_STREAM_1, 6);
+        assertTopicSize(OUTPUT_STREAM_2, 3);
+    }
+
+    @Test
+    public void shouldAllowForNamedTopologiesToStartPaused() throws Exception {
+        streamsNamedTopologyWrapper = new KafkaStreamsNamedTopologyWrapper(props());
+        final NamedTopologyBuilder builder1 = getNamedTopologyBuilder1();
+        final NamedTopologyBuilder builder2 = getNamedTopologyBuilder2();
+
+        streamsNamedTopologyWrapper.pauseNamedTopology(TOPOLOGY1);
+        streamsNamedTopologyWrapper.start(asList(builder1.build(), builder2.build()));
+        waitForApplicationState(singletonList(streamsNamedTopologyWrapper), State.RUNNING, STARTUP_TIMEOUT);
+
+        assertFalse(streamsNamedTopologyWrapper.isPaused());
+        assertTrue(streamsNamedTopologyWrapper.isNamedTopologyPaused(TOPOLOGY1));
+        assertFalse(streamsNamedTopologyWrapper.isNamedTopologyPaused(TOPOLOGY2));
+
+        produceToInputTopics(INPUT_STREAM_1, STANDARD_INPUT_DATA);
+        produceToInputTopics(INPUT_STREAM_2, STANDARD_INPUT_DATA);
+
+        awaitOutput(OUTPUT_STREAM_2, 3, COUNT_OUTPUT_DATA);
+        assertTopicSize(OUTPUT_STREAM_1, 0);
+
+        streamsNamedTopologyWrapper.pause();
+        assertTrue(streamsNamedTopologyWrapper.isPaused());
+        assertTrue(streamsNamedTopologyWrapper.isNamedTopologyPaused(TOPOLOGY1));
+        assertTrue(streamsNamedTopologyWrapper.isNamedTopologyPaused(TOPOLOGY2));
+
+        produceToInputTopics(INPUT_STREAM_1, STANDARD_INPUT_DATA);
+        produceToInputTopics(INPUT_STREAM_2, STANDARD_INPUT_DATA);
+
+        assertNoLag(streamsNamedTopologyWrapper);
+
+        waitUntilStreamsHasPolled(streamsNamedTopologyWrapper, 2);
+        assertTopicSize(OUTPUT_STREAM_1, 0);
+        assertTopicSize(OUTPUT_STREAM_2, 3);
+
+        streamsNamedTopologyWrapper.resumeNamedTopology(TOPOLOGY1);
+        assertFalse(streamsNamedTopologyWrapper.isPaused());
+        assertFalse(streamsNamedTopologyWrapper.isNamedTopologyPaused(TOPOLOGY1));
+        assertTrue(streamsNamedTopologyWrapper.isNamedTopologyPaused(TOPOLOGY2));
+
+        awaitOutput(OUTPUT_STREAM_1, 3, COUNT_OUTPUT_DATA2);
+        assertTopicSize(OUTPUT_STREAM_1, 3);
+        assertTopicSize(OUTPUT_STREAM_2, 3);
+    }
+
+    @Test
+    public void pauseResumehouldWorkAcrossInstances() throws Exception {
```

Comment:
According to the builds, this test seems flaky. I think the reason is that the cache (in Streams, not the RocksDB cache) of the state stores is not set to zero. When the cache is not set to zero, the number of results that are send downstreams is not deterministic, because some intermediate results might be sent downstream and some not. I tried to set the cache to zero and was able to run 100+ times the test in a row without failure whereas with the cache > 0, the test failed much earlier.
Note that by changing the cache size also the results that you verify change.

Reply:
Thanks.  I was having trouble shifting through the build info.  I've updated the test to set the state store size to 0.  

I'll watch the CI jobs to see if they are green after my last push.  (They seemed to be ok as I was working today.)